recipes/protobuf/all/conanfile.py

```diff
@@ -94,69 +92,75 @@ def _configure_cmake(self):
             self._cmake.definitions["protobuf_WITH_ZLIB"] = self.options.with_zlib
             self._cmake.definitions["protobuf_BUILD_TESTS"] = False
             self._cmake.definitions["protobuf_BUILD_PROTOC_BINARIES"] = True
-            self._cmake.definitions["protobuf_BUILD_LIBPROTOC"] = True
+            if tools.Version(self.version) >= "3.14.0":
+                self._cmake.definitions["protobuf_BUILD_LIBPROTOC"] = True
             if self._can_disable_rtti:
                 self._cmake.definitions["protobuf_DISABLE_RTTI"] = not self.options.with_rtti
-            if self.settings.compiler == "Visual Studio":
-                self._cmake.definitions["protobuf_MSVC_STATIC_RUNTIME"] = "MT" in str(self.settings.compiler.runtime)
+            if self.settings.compiler.get_safe("runtime"):
+                self._cmake.definitions["protobuf_MSVC_STATIC_RUNTIME"] = str(self.settings.compiler.runtime) in ["MT", "MTd", "static"]
             self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
     def _patch_sources(self):
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
 
-        find_protoc = """
-
-# Find the protobuf compiler within the paths added by Conan, for use below.
-find_program(PROTOC_PROGRAM protoc PATHS ENV PATH NO_DEFAULT_PATH)
-if(NOT PROTOC_PROGRAM)
-    set(PROTOC_PROGRAM "protoc")
```

Comment:
It's not clear to me how this is not conflicting with:
https://github.com/conan-io/conan-center-index/commit/369121c6093c8a6dbd51ea7f90ba8d37dcd3056f#diff-5dfb647bf8fb82769ddb94143d6b47475c5d4d2db2346115ee6b1991a6db6445R112
What will the merge look like?

Reply:
It's removed. This PR adds `libprotobuf::protoc` target, but it's not very clear for me if it works with two profiles.

Reply:
I have pasted some commands below with one way you can test it.
The dual profile build doesn't seem to work.
CMake will find the wrong `protoc` (the one from host context instead of the one from build context).

Command:
```bash
conan create . 3.15.5@foo/testing \
    --profile:host android-ndk-r22-api-21-arm64-v8a-clang-c++_static.txt \
    --profile:build default \
    --build=protobuf
```

android-ndk-r22-api-21-arm64-v8a-clang-c++_static.txt:
```
[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.libcxx=c++_static
compiler.version=11
os=Android
os.api_level=21
[build_requires]
android-ndk/r22
[options]
[env]
```

Result:
```
protobuf/3.15.5@foo/testing (test package): Calling build()
...
[1/4] Running cpp protocol buffer compiler on addressbook.proto
FAILED: addressbook.pb.h addressbook.pb.cc
...
/bin/sh: 1: x/.conan/data/protobuf/3.15.5/dirac/testing/package/475278d37419b9e365c649e594b7c9a419a343bb/bin/protoc: Exec format error
```

Reply:
Ok so maybe we could keep this `find_program`, and use this path for `protobuf::protoc` properties.

Reply:
Yeah, could work. I'm happy to try if you push something later :)

Reply:
Could you test with https://github.com/conan-io/conan-center-index/pull/4776/commits/a0f4c761d7f44167961a272a407d215a439e086c please?

Reply:
Seems to work just fine. I'm happy with it :)